src/job.rs

```diff
@@ -639,7 +639,21 @@ impl Job {
         let msg = job_try!(Message::load_from_db(context, MsgId::new(self.foreign_id)).await);
 
         let folder = msg.server_folder.as_ref().unwrap();
-        match imap.set_seen(context, folder, msg.server_uid).await {
+
+        let result = if msg.server_uid == 0 {
+            // The message is moved or deleted by us.
```

Comment:
side-question: Can we distinguish between moved and deleted?

Reply:
Deleted messages have empty `server_folder` and zero `server_uid`, but moved messages have old `server_folder` and zero `server_uid`. But I wouldn't rely on it.